src/dowel/csv_output.py

```diff
@@ -42,12 +40,23 @@ def record(self, data, prefix=''):
                 self._writer.writeheader()
 
             if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
+                              
+             #save the old file logs in a variable
+                with open(self._log_file.name) as log_in_csv:
+                    old_data = csv.DictReader(log_in_csv)
```

Comment:
`DictReader` actually reads the data from the CSV on demand, so this likely only works because of file buffering. Can you think of how to fix it?

Reply:
Yes DictReader is reading data whenever inconsistency found and used to save the old data in a variable and merge it with new data.